Allow multiple targets on command line

n-oden opened the pull request:
This addresses https://github.com/incu6us/goimports-reviser/issues/96

- iterate over the list of remaining arguments treating each as an origin path

- defer printing deprecation warnings until after all paths are processed

- print deprecation warnings to stderr, not stdout

n-oden commented:
@incu6us I've updated this against current upstream/master and fixed the conflict

n-oden commented:
Is the codecov test advisory, or are you asking for test coverage going forward? There's no current test set for main.go, but I suppose we could make one...

ccoVeille commented on main.go:
```diff
@@ -277,21 +277,24 @@ func main() {
-		os.Exit(1)
+	for _, originPath = range originPaths {
+		if err := validateRequiredParam(originPath); err != nil {
+			fmt.Printf("%s\n\n", err)
```
You are using log.Fatal to handle other errors

log package use stdErr when nothing is configured

https://cs.opensource.google/go/go/+/refs/tags/go1.23.3:src/log/log.go;l=87


Here your code use, stdOut

I would suggest you using this

```suggestion
			log.Printf("%s\n\n", err)
```

ccoVeille commented on main.go:
```diff
@@ -277,21 +277,24 @@ func main() {
+	for _, originPath = range originPaths {
+		if err := validateRequiredParam(originPath); err != nil {
+			fmt.Printf("%s\n\n", err)
+			printUsage()
+			os.Exit(1)
```
You are looping on argument to validate them.

Maybe you could report an error for each argument

Then after the loop printusage and exit, when at least an error is found

n-oden replied:
Actually, given that `validateRequiredParam` is currently only checking to see if stdin exists in the cast of `originPaths[0]` being `-` I think this entire loop can be removed and the call to `validateRequiredParam` moved up into the previous block. @incu6us I'm going to make this change but let me know what you think?

incu6us replied:
@n-oden yep, agree

incu6us commented:
> @incu6us I've updated this against current upstream/master and fixed the conflict

could you update it once again?

n-oden commented:
@incu6us @ccoVeille addressed the comments, rebased on current upstream/master, and while I was at it spotted another issue: the call to helper.DetermineProjectName needed to be moved inside the loop over `originPaths` since different paths might have different project names.

ccoVeille commented:
Great. Let's waif for @incu6us feedbacks then

ccoVeille commented on main.go:
```diff
@@ -285,21 +285,20 @@ func main() {
-		os.Exit(1)
+	if len(originPaths) == 0 || (len(originPaths) == 1 && originPaths[0] == "-") {
+		originPaths[0] = reviser.StandardInput
+		if err := validateRequiredParam(originPaths[0]); err != nil {
+			log.Printf("%s\n\n", err)
+			printUsage()
+			os.Exit(1)
+		}
 	}
```
Shouldn't be this?


```suggestion
 	}
	if err := validateRequiredParam(originPaths[0]); err != nil {
		log.Printf("%s\n\n", err)
		printUsage()
		os.Exit(1)
	}
```

n-oden replied:
I think I'm possibly misunderstanding something here. `validateRequiredParam()` will always return nil if the passed argument is anything other than `reviser.StandardInput`:

```go
func validateRequiredParam(filePath string) error {
	if filePath == reviser.StandardInput {
		stat, _ := os.Stdin.Stat()
		if stat.Mode()&os.ModeNamedPipe == 0 {
			// no data on stdin
			return errors.New("no data on stdin")
		}
	}
	return nil
}
```

So I think there's no point in calling it unless we know that we're operating on stdin, which would only happen if we find ourselves on the inside of `if len(originPaths) == 0 || (len(originPaths) == 1 && originPaths[0] == "-") {`

ccoVeille commented on main.go:
```diff
@@ -344,116 +343,117 @@ func main() {
+		originProjectName, err := helper.DetermineProjectName(projectName, originPath, helper.OSGetwdOption)
+		if err != nil {
+			log.Printf("Could not determine project name for path %s: %s\n\n", originPath, err)
+			printUsage()
+			os.Exit(1)
```
I would add an `printUsageAndFail` to do that, you are repeating it everywhere

ccoVeille reviewed (commented):
Some minor remarks, open to debate

ccoVeille commented on main.go:
```diff
@@ -206,6 +206,14 @@ func printUsage() {
+	if err != nil {
+		log.Fatalf("%s", err)
+	}
+	os.Exit(0)
```
```suggestion
	os.Exit(1)
```

ccoVeille replied:
Wow, indeed. Clever

n-oden commented:
Thanks to all of you for getting my PRs merged and released!
